### DISCUSSION TITLE
Contextual menu for QT version of DC

### DISCUSSION THREAD

**Discussion by savornicesei** [doublecmd/doublecmd#1051] [General]
Hi all,

I'm using the QT version of DC on openSUSE Tumbleweed and the contextual menu is not as rich as Dolphin's. 
Is it possible to enrich it (as it's possible in Dolphin)? 
Could you share some details about how the contextual menu is loaded?

I believe the code to create the context menu is in [ushellcontextmenu.pas](https://github.com/doublecmd/doublecmd/blob/master/src/platform/unix/ushellcontextmenu.pas) file and there is also a [upython.pas](src/platform/unix/upython.pas) and [urabbitvcs.pas](https://github.com/doublecmd/doublecmd/blob/master/src/platform/unix/urabbitvcs.pas) that make it possible to integrate rabbitvcs in DC. Would it be possible to expand the menu using a python script? How is the [rabbitvcs py ](https://github.com/doublecmd/doublecmd/blob/master/scripts/rabbit-vcs.py) script used by DC?

Thank you.

All the best,
Simo

**Comment by alexx2000** [comment]
Hi,

Directly you can change only `Actions` submenu via [File associations](https://doublecmd.github.io/doc/en/configuration.html#ConfigAssociations).

> Would it be possible to expand the menu using a python script? How is the [rabbitvcs py ](https://github.com/doublecmd/doublecmd/blob/master/scripts/rabbit-vcs.py)script used by DC?

Technically you can replace `rabbit-vcs.py` with your code and generate needed menu items. DC calls `GetContextMenu` function and pass selected files to it, call `GetMenu` method to get menu items, call `Execute` method when user choose some menu item.
